modules/shmelev_i_graham_algorithm/src/graham_algorithm.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 Shmelev Ivan
+
+#include "include/graham_algorithm.h"
+
+
+
+int GrahamScan::orientation(const std::pair<double, double>& p,
+                             const std::pair<double, double>& q,
+                             const std::pair<double, double>& r) {
+    double val = (q.second - p.second) * (r.first - q.first) -
+                        (q.first - p.first) * (r.second - q.second);
+    if (val == 0) return COLLINEAR;
+    return (val > 0) ? LEFT_TURN : RIGHT_TURN;
+}
+
+std::vector<std::pair<double, double>> GrahamScan::sortPoints
+                        (std::vector<std::pair<double, double>>& points) {
+    int n = points.size();
```

Comment:
Избегайте числовые конверсии
Аналогично в других местах
```suggestion
    size_t n = points.size();
```